pkg/controller/quota/quota.go

```diff
@@ -169,6 +178,25 @@ func addStorage(appInstance *v1.AppInstance, quotaRequest *adminv1.QuotaRequestI
 	return nil
 }
 
+func defaultVolumeSize(appInstance *v1.AppInstance, name string) resource.Quantity {
+	result := *v1.DefaultSize // Safe to dereference because it is statically set in the v1 package.
+
+	// If the volume has a default size set, use that on status.Defaults, use that. Otherwise, if the
+	// VolumeSize default is set on status.Defaults, use that.
+	if defaultVolume, set := appInstance.Status.Defaults.Volumes[name]; set {
+		// We do not expect this to ever fail because VolumeClasses have their sizes validated. However,
+		// if it does fail, we'll just return the default size.
+		parsedQuantity, err := resource.ParseQuantity(string(defaultVolume.Size))
+		if err == nil {
+			result = parsedQuantity
```

Comment:
This felt like the right behavior to me if a size failed to parse, but let me know if something else makes more sense.